pkg/controllers/provisioningv2/rke2/common.go

```diff
@@ -0,0 +1,163 @@
+package rke2
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	capicontrollers "github.com/rancher/rancher/pkg/generated/controllers/cluster.x-k8s.io/v1beta1"
+	"github.com/rancher/wrangler/pkg/condition"
+	"github.com/rancher/wrangler/pkg/generic"
+	"github.com/rancher/wrangler/pkg/name"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	capierrors "sigs.k8s.io/cluster-api/errors"
+)
+
+const (
+	AddressAnnotation = "rke.cattle.io/address"
+	ClusterNameLabel  = "rke.cattle.io/cluster-name"
+	// ClusterSpecAnnotation is used to define the cluster spec used to generate the rkecontrolplane object as an annotation on the object
+	ClusterSpecAnnotation      = "rke.cattle.io/cluster-spec"
+	ControlPlaneRoleLabel      = "rke.cattle.io/control-plane-role"
+	DrainAnnotation            = "rke.cattle.io/drain-options"
+	DrainDoneAnnotation        = "rke.cattle.io/drain-done"
+	EtcdRoleLabel              = "rke.cattle.io/etcd-role"
+	InitNodeLabel              = "rke.cattle.io/init-node"
+	InitNodeMachineIDDoneLabel = "rke.cattle.io/init-node-machine-id-done"
+	InitNodeMachineIDLabel     = "rke.cattle.io/init-node-machine-id"
+	InternalAddressAnnotation  = "rke.cattle.io/internal-address"
+	JoinURLAnnotation          = "rke.cattle.io/join-url"
+	LabelsAnnotation           = "rke.cattle.io/labels"
+	MachineIDLabel             = "rke.cattle.io/machine-id"
+	MachineNameLabel           = "rke.cattle.io/machine-name"
+	MachineNamespaceLabel      = "rke.cattle.io/machine-namespace"
+	MachineRequestType         = "rke.cattle.io/machine-request"
+	MachineUIDLabel            = "rke.cattle.io/machine"
+	NodeNameLabel              = "rke.cattle.io/node-name"
+	PlanSecret                 = "rke.cattle.io/plan-secret-name"
+	PostDrainAnnotation        = "rke.cattle.io/post-drain"
+	PreDrainAnnotation         = "rke.cattle.io/pre-drain"
+	RoleLabel                  = "rke.cattle.io/service-account-role"
+	SecretTypeMachinePlan      = "rke.cattle.io/machine-plan"
+	TaintsAnnotation           = "rke.cattle.io/taints"
+	UnCordonAnnotation         = "rke.cattle.io/uncordon"
+	WorkerRoleLabel            = "rke.cattle.io/worker-role"
+
+	CattleOSLabel = "cattle.io/os"
+
+	RKEMachineAPIVersion = "rke-machine.cattle.io/v1"
+
+	Provisioned = condition.Cond("Provisioned")
+	Ready       = condition.Cond("Ready")
+	Waiting     = condition.Cond("Waiting")
+	Pending     = condition.Cond("Pending")
+	Removed     = condition.Cond("Removed")
+
+	RuntimeK3S  = "k3s"
+	RuntimeRKE2 = "rke2"
+)
+
+var (
+	ErrNoMachineOwnerRef = errors.New("no machine owner ref")
+)
+
+func MachineStateSecretName(machineName string) string {
+	return name.SafeConcatName(machineName, "machine", "state")
+}
+
+func GetMachineOwner(machineCache capicontrollers.MachineCache, obj metav1.Object) (*capi.Machine, error) {
```

Comment:
This function name is confusing to me -- it appears that it gets a machine that owns the passed in `obj`, but when I read the function name I assume it is getting the owner of the machine.

Reply:
Will update.